ginga/misc/plugins/LineProfile.py

```diff
@@ -0,0 +1,425 @@
+
+        except KeyError:
+            # Add canvas layer
+            self.fitsimage.add(self.canvas, tag=self.layertag)
+
+        self.resume()
+
+    def pause(self):
+        self.canvas.ui_setActive(False)
+
+    def resume(self):
+        self.canvas.ui_setActive(True)
+        self.redo()
+
+    def stop(self):
+        self.canvas.ui_setActive(False)
+        try:
+            self.fitsimage.deleteObjectByTag(self.layertag)
+        except:
+            pass
+
+    def redo(self):
+        # Get image being shown
+        self.image = self.fitsimage.get_image()
+        if self.image is None:
+            return
+        self.wd, self.ht = self.image.get_size()
+
+        self.redraw_mark()
+
+    def _plot(self, mark=None):
+        # Transpose array for easier slicing
+        mddata = self.image.get_mddata().T
+        naxes = mddata.ndim
+
+        if self.selected_axis:
+            axis_data = self.get_axis(self.selected_axis)
+            slice_obj = self._slice(naxes, mk=mark)
+
+            self.clear_plot()
+            self.plot.plot(axis_data, mddata[slice_obj])
+
+    def _slice(self, naxes, mk):
+        # Build N-dim slice
+        slice_obj = [0] * naxes
+
+        # For axes 1 and 2
+        if mk is not None:
+            slice_obj[0] = self.mark_data_x[mk]
+            slice_obj[1] = self.mark_data_y[mk]
+
+        # For axis > 3
+        for i in xrange(2, naxes):
+            slice_obj[i] = self.image.revnaxis[i-2] + 1
+
+        # Slice enabled axis
+        slice_obj[self.selected_axis-1] = slice(None, None, None)
+
+        return slice_obj
+
+    def get_axis(self, i):
+        try:
+            header = self.image.get_header()
+            axis = header.get('CRVAL%d' % i) + \
+                   np.arange(0, header.get('NAXIS%d' % i), 1) * \
+                   header.get('CDELT%d' % i)
+            return axis
+        except Exception as e:
+            errmsg = "Error loading axis %d: %s" % (i, str(e))
+            self.logger.error(errmsg)
+            self.fv.error(errmsg)
+
+    def clear_plot(self):
+        self.plot.clear()
+        self.plot.fig.canvas.draw()
+
+    ### MARK FEATURE LOGIC ###
+
+    def btndown_cb(self, canvas, event, data_x, data_y):
+        # Exclude points outside boundaries
+        if not 0 <= data_x < self.wd or not 0 <= data_y < self.ht:
+            self.clear_plot()
+            return
+
+        if not self.mark_selected:
+            self.mark_data_x.append(data_x)
+            self.mark_data_y.append(data_y)
+            self.add_mark(data_x, data_y)
+
+            self.del_btn.set_enabled(True)
+            self.del_all_btn.set_enabled(True)
+        return True
+
+    def update(self, canvas, event, data_x, data_y):
+        tag = self.mark_selected
+        if tag is None:
+            return
+        idx = int(tag.strip('mark'))
+        obj = self.canvas.getObjectByTag(tag)
+        obj.move_to(data_x, data_y)
```

Comment:
You probably want to change this to:
```python
obj.move_to(data_x+5, data_y)
```
because you are moving the compound object *center*.